nixos/modules/services/network-filesystems/openafs/server.nix

```diff
@@ -62,6 +62,68 @@ let
     k5start.command = (lib.getBin pkgs.kstart) + "/bin/k5start";
   } // cfg.roles.backup.fabsExtraConfig));
 
+  asetkeyScript =
+    let
+      cfgAuth = cfg.authentication;
+
+      # Values taken from krb5/krb5.h
+      krb5Enctypes = {
+        des-cbc-crc = 1;
+        des-cbc-md4 = 2;
+        des-cbc-md5 = 3;
+        des-cbc-raw = 4;
+        des3-cbc-sha = 5;
+        des3-cbc-raw = 6;
+        des-hmac-sha1 = 8;
+        dsa-sha1-cms = 9;
+        md5-rsa-cms = 10;
+        sha1-rsa-cms = 11;
+        rc2-cbc-env = 12;
+        rsa-env = 13;
+        rsa-es-oaep-env = 14;
+        des3-cbc-env = 15;
+        des3-cbc-sha1 = 16;
+        aes128-cts-hmac-sha1-96 = 17;
+        aes256-cts-hmac-sha1-96 = 18;
+        aes128-cts-hmac-sha256-128 = 19;
+        aes256-cts-hmac-sha384-192 = 20;
+        arcfour-hmac = 23;
+        arcfour-hmac-exp = 24;
+        camellia128-cts-cmac = 25;
+        camellia256-cts-cmac = 26;
+      };
+
+      # Stringified Awk array et["<type name>"] = <type id> of known encryption types
+      enctypeAwkArray = lib.concatStringsSep "\n"
+        (lib.mapAttrsToList (t: n: ''et["${t}"] = ${toString n}'') krb5Enctypes);
+
+      # Awk script that, given some Kerberos 5 key metadata in the format
+      # printed by ktutil's list command, extracts the key version number (KVNO)
+      # and encryption type of each key, converts the encryption type into a
+      # numeric value understood by asetkey, and uses this information to
+      # initialize the KeyFileExt database for OpenAFS.
+      awkAsetkeyScript = pkgs.writeText "asetkey-script.awk" ''
+        BEGIN {
+          ${enctypeAwkArray}
+        }
+
+        # Skip over the table header and only match the configured service
+        # principal
+        (NR > 2) && ($3 == "${cfgAuth.principal}") {
+            kvno = $2
+            princ = $3
+            # The encryption type returned by ktutil is surrounded by parentheses
+            enctype = et[substr($4, 2, length($4)-2)]
+
+            # Import each key into the KeyFileExt database
+            system("${openafsBin}/bin/asetkey add rxkad_krb5 " kvno " " enctype " ${cfgAuth.keytab} " princ)
+        }
+      '';
+    in
+    pkgs.writeShellScript "asetkey-script.sh" ''
+      echo -e "rkt ${cfgAuth.keytab}\nlist -e" | ${pkgs.krb5}/bin/ktutil | \
```

Comment:
Document that krb5 `ktutil` must be used here in a comment above the script to prevent future refactoring. (One could be tempted to point to `config.security.krb5.package`.)